src/main_v2.py

```diff
@@ -106,65 +124,84 @@ async def scrape_data(
-            player: Player
-            hiscore: dict
-        except InvalidResponse as _:
-            error_type = type(error)
-            sleep_time = max(error_count * 2, 5)
-            await player_send_queue.put(item=player.dict())
-            await asyncio.sleep(sleep_time)
-            continue
-        except (ClientResponseError, ClientHttpProxyError) as error:
-            session = ClientSession(
-                timeout=ClientTimeout(total=app_config.SESSION_TIMEOUT)
-            )
-            error_type = type(error)
-            sleep_time = 35
-            logger.error(
-                f"{name} - {error_type.__name__}: {str(error)} - {error_count=} - {player.name=}"
-            )
-            await player_send_queue.put(item=player.dict())
-            await asyncio.sleep(sleep_time)
-            continue
-        except Exception as error:
-            error_type = type(error)
-            sleep_time = max(error_count * 2, 5)
-            logger.error(
-                {
-                    "name": name,
-                    "error_type": error_type.__name__,
-                    "error": error,
-                    "error_count": error_count,
-                    "player_name": player.name,
-                }
-            )
-            tb_str = traceback.format_exc()
-            logger.error(f"{error}, \n{tb_str}")
-            await player_send_queue.put(item=player.dict())
-            await asyncio.sleep(sleep_time)
-            continue
+    async with ClientSession(
+        timeout=ClientTimeout(total=app_config.SESSION_TIMEOUT)
+    ) as session:
+        while True:
+            if shutdown_event.is_set():
+                break
+            if player_receive_queue.empty():
+                # logger.info(f"{name=} - receive queue is empty")
+                await asyncio.sleep(5)
+                continue
+
+            if error_count > 5:
+                logger.warning(f"high error count: {error_count}, killing task")
+                break
+
+            player = await player_receive_queue.get()
+            player_receive_queue.task_done()
+
+            player = Player(**player)
+            # logger.info(f"Scraping data for player: {player.name}")
+            try:
+                player, hiscore = await scraper.lookup_hiscores(player, session)
+                player: Player
+                hiscore: dict
+                # logger.info(f"Scraped data for player: {player.name}")
+                scrape_counter += 1
+                logger.debug(f"scrape_counter: {scrape_counter}")  # Add this line
+                if scrape_counter == 100:
+                    logger.debug("scrape_counter reached 100")  # Add this line
+                    end_time = time.time()
+                    logger.info(
+                        f"Scraped 100 players in {end_time - start_time} seconds"
+                    )
+                    start_time = time.time()
+                    scrape_counter = 0
```

Comment:
you really don't want this, every proxy will run this at the same time (#250 proxies), the desired speed should be about 500.000 scrapes an hour, in the past we've seen significant slowdoing logging this every line